### PR TITLE
Scrub per group comments

### PR DESCRIPTION
**Pull request by tammyleino**
Scrubbing docs for structure.
Signed-off-by: Tammy Leino <tammy_leino@mentor.com>

### CODE REVIEW

**Comment by tammyleino** [comment]
Per @wmamills comments at OA-syst-ref meeting 16-Nov-22

     Current work (hide for now)
     Samples and Demo
	TOC on this page should not inclzude subheads
     Maintenance
         change title to "Contributing to OpenAMP"
     Meeting Notes (hide/delete)
     Future work (hide)
OpenAMP Protocol Details
     Porting Guide (move to OpenAMP Library UG)
     Using OpenAMP CI
	move to Samples & Demos under name of
	"OpenAMP Demo Docker images"
OpenAMP Library Users Guide (new section)
     Porting Guide

**Also created separate index page for data structures to condense content and created separate index page for design docs and moved to protocol details section.

File: demos/docker_images.rst [hunk 1]
```diff
@@ -1,6 +1,6 @@
-================
-Using OpenAMP CI
-================
+==========================
+OpenAMP Demo Docker images
+==========================
 
 The OpenAMP CI solution is just in it's infancy at the moment and this doc is a work-in-progress. Please keep it up to date and don't keep historical information (that is what wiki history is for).
```
Comment (arnopo) [inline demos/docker_images.rst:5]: does make sense to speak about "OpenAMP CI solution"?

File: openamp/contributing.rst [hunk 2]
```diff
@@ -1,8 +1,8 @@
 .. _openamp-maintenance-work-label:
 
-===========
-Maintenance
-===========
+=======================
+Contributing to OpenAMP
+=======================
```
Comment (arnopo) [inline openamp/contributing.rst:5]: the sub title below seems useless

Comment (tammyleino) [reply openamp/contributing.rst:5]: @arnopo Agreed

File: demos/docker_images.rst (hunk 1 above)
Comment (tammyleino) [reply demos/docker_images.rst:5]: @wmamills @edmooring ?

File: openamp/contributing.rst (hunk 2 above)
Comment (tammyleino) [reply openamp/contributing.rst]: @arnopo Fixed.

File: demos/docker_images.rst (hunk 1 above)
Comment (tammyleino) [reply demos/docker_images.rst:5]: @wmamills Has taken on the action to scrub this content.  This commit is just to address his comment to move and rename:

     Using OpenAMP CI
	move to Samples & Demos under name of
	"OpenAMP Demo Docker images"
	I will be scrub owner for this page

**Pull request merged**